csfieldguide/static/interactives/scene-editor/js/scene-editor.js

```diff
@@ -0,0 +1,694 @@
+
+  $("#add-object").click(newObject);
+  $("#apply-transformation").click(applyTransformation);
+
+  $("#colour-input").val('');
+  $("#name-input").val('');
+});
+
+/**
+ * Creates the new scene; skybox, lighting, camera and the initial object
+ */
+function init() {
+  // Cameras
+  camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 100000 );
+  camera.position.set( 1000, 500, 1000 );
+  cameraCube = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 100000 );
+  camera.lookAt( new THREE.Vector3(0, 0, 0) );
+  // Scene
+  scene = new THREE.Scene();
+  sceneCube = new THREE.Scene();
+  // Lights
+  var ambient = new THREE.AmbientLight( 0xffffff );
+  scene.add( ambient );
+  // Textures
+  var urls = [ 
+    image_base_path + "posx.jpg", image_base_path + "negx.jpg",
+    image_base_path + "posy.jpg", image_base_path + "negy.jpg",
+    image_base_path + "posz.jpg", image_base_path + "negz.jpg" 
+  ];
+  textureCube = new THREE.CubeTextureLoader().load( urls );
+  textureCube.format = THREE.RGBFormat;
+  textureCube.mapping = THREE.CubeReflectionMapping;
+  textureCube.encoding = THREE.sRGBEncoding;
+
+  var cubeShader = THREE.ShaderLib[ "cube" ];
+  var cubeMaterial = new THREE.ShaderMaterial( {
+    fragmentShader: cubeShader.fragmentShader,
+    vertexShader: cubeShader.vertexShader,
+    uniforms: cubeShader.uniforms,
+    depthWrite: false,
+    side: THREE.BackSide
+  } );
+  cubeMaterial.uniforms[ "tCube" ].value = textureCube;
+  Object.defineProperty( cubeMaterial, 'map', {
+    get: function () {
+      return this.uniforms.tCube.value;
+    }
+  } );
+  // Skybox
+  cubeMesh = new THREE.Mesh( new THREE.BoxBufferGeometry( 100, 100, 100 ), cubeMaterial );
+  sceneCube.add( cubeMesh );
+  // Sphere object
+  isStartingShape = true;
+  var sphereMaterial = new THREE.MeshLambertMaterial( { envMap: textureCube } );
+  addObject('sphere', sphereMaterial, '');
+  // Camera orbit pointer
+  addObject('tinyaxis', null, null);
+  cameraPointer = scene.getObjectByName( CAMERA_POINTERID )
+  //
+  renderer = new THREE.WebGLRenderer();
+  renderer.autoClear = false;
+  renderer.setPixelRatio( window.devicePixelRatio );
+  renderer.setSize( window.innerWidth, window.innerHeight );
+  var container = document.getElementById('scene');
+  container.appendChild( renderer.domElement );
+  renderer.gammaOutput = true;
+  //
+  controls = new OrbitControls( camera, renderer.domElement );
+  controls.minDistance = 500;
+  controls.maxDistance = 10000;
+
+  // Grid
+  var size = 10000;
+  var divisions = 10;
+  var colorCenterLine = 0xffffff;
+  var colorGrid = 0xffffff;
+  var gridHelper = new THREE.GridHelper( size, divisions, colorCenterLine, colorGrid );
+  scene.add( gridHelper );
+  // Axes
+  addAxes(size);
+
+  window.addEventListener( 'resize', onWindowResize, false );
+}
+
+/**
+ * Rescales the scene to the size of the window
+ */
+function onWindowResize() {
```

Comment:
This probably doesn't matter for an MVP, but i'll put a note anyway. The scene doesn't stay in proportion when the window is resized, e.g the sphere appears stretched:

![image](https://user-images.githubusercontent.com/16066822/66357501-a6f1ad00-e9cb-11e9-954f-e3f94546c2ea.png)

Reply:
Hmm, that's unique to chrome